### PR TITLE
resource/aws_ses_event_destination: Add support for SNS destinations

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #1697

```
terraform-provider-aws [master●] % acctests aws TestAccAWSSESEventDestination_basic
=== RUN   TestAccAWSSESEventDestination_basic
--- PASS: TestAccAWSSESEventDestination_basic (159.86s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	159.898s
```

### CODE REVIEW

File: aws/resource_aws_ses_event_destination_test.go [hunk 1]
```diff
@@ -156,6 +158,10 @@ data "aws_iam_policy_document" "fh_felivery_document" {
     }
 }
 
+resource "aws_sns_topic" "ses_destination" {
+  name = "ses-destination-test"
```
Comment (radeksimko) [inline aws/resource_aws_ses_event_destination_test.go]: Do you mind randomizing this name here?

File: aws/resource_aws_ses_event_destination_test.go [hunk 2]
```diff
@@ -178,10 +184,22 @@ resource "aws_ses_event_destination" "cloudwatch" {
...
-  cloudwatch_destination = {
+  cloudwatch_destination {
     default_value = "default"
 	dimension_name = "dimension"
 	value_source = "emailHeader"
   }
 }
+
+resource "aws_ses_event_destination" "sns" {
+  name = "event-destination-sns",
```
Comment (radeksimko) [inline aws/resource_aws_ses_event_destination_test.go]: Likewise ^ do you mind randomizing it?

File: aws/resource_aws_ses_event_destination_test.go [hunk 3]
```diff
@@ -26,6 +26,8 @@ func TestAccAWSSESEventDestination_basic(t *testing.T) {
 						"aws_ses_event_destination.kinesis", "name", "event-destination-kinesis"),
 					resource.TestCheckResourceAttr(
 						"aws_ses_event_destination.cloudwatch", "name", "event-destination-cloudwatch"),
+					resource.TestCheckResourceAttr(
+						"aws_ses_event_destination.sns", "name", "event-destination-sns"),
```
Comment (radeksimko) [inline aws/resource_aws_ses_event_destination_test.go]: Nitpick - but do you mind adding check for `sns_destination.0.topic_arn` as that is the field being added and that's the one that should also be tested?

**Comment by stack72** [comment]
@radeksimko so I looked at this and the reason the original tests are not actually randomized is that there is nothing happening in the Read func - therefore, when we check that a destroy happens, we are actually checking the hardcoded SES destination to make sure it no longer exists

Thoughts?

P.

**Comment by caJaeger** [comment]
@radeksimko Is this dead in the water? I would love to see this in the next release...

**Comment by sefaYavuz** [comment]
+1 need this!

**Comment by pccowboy** [comment]
@stack72 I have what might be valid tests for what @radeksimko has asked for - how best to get them into this PR?

**Comment by radeksimko** [comment]
There are other things to improve on that resource - e.g. turning all the `MaxItems: 1` `TypeSet`s into `TypeList`s to make nested fields easier to reference in tests and elsewhere, but that's certainly outside of scope of this PR.

Merging now...

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

File: aws/resource_aws_ses_event_destination.go [hunk 4]
```diff
@@ -53,20 +53,21 @@ func resourceAwsSesEventDestination() *schema.Resource {
 				Type:          schema.TypeSet,
 				Optional:      true,
 				ForceNew:      true,
-				ConflictsWith: []string{"kinesis_destination"},
+				MaxItems:      1,
```
Comment (bflad) [inline aws/resource_aws_ses_event_destination.go]: Interestingly enough, this resource actually did appear to support multiple `cloudwatch_destination` previously, even if it wasn't appropriately acceptance tested. This change will likely be reverted in #6690

**Pull request merged**